src/transformers/configuration_fsmt.py

```diff
@@ -126,9 +126,9 @@ class FSMTConfig(PretrainedConfig):
     # update the defaults from config file
     def __init__(
         self,
-        langs,
-        src_vocab_size,
-        tgt_vocab_size,
+        langs=["en", "de"],
+        src_vocab_size=42024,
+        tgt_vocab_size=42024,
```

Comment:
The suggested vocab size defaults make no sense w/o the corresponding vocab. Just as well set them to 1000 if defaults are now required. I remember some common tests fails if vocab size is less than 1000, so that is probably a good default.

Reply:
ok great! Thanks for the tip :-)

Reply:
Sorry, actually I don't really see why 42024 makes no sense - could you explain a bit?

Reply:
Is this a set of defaults for the sake of default, or are we configuring a very specific tiny model as a default that is actually a working model?

Reply:
If you're using some real model as a default, then yes, you want the actual numbers. If these are just random numbers, then why would you want to set it to 42024?

Reply:
The "makes no sense" part is about a user reading model's default and scratching their head about why this particular number, or why these languages? Normally default values make certain sense. That's why I removed the defaults, since any defaults make no sense here.

The model can't do anything sensible with `en/de` without the specific model trained on this pair to support it. It can't do anything with the vocab_size of 42024 w/o that vocab. I hope this helps explaining it.

Reply:
Sorry answered above - yeah I'm in favor of actual numbers -> I don't think they should be random numbers.

Reply:
Thank you for explaining your point of view, @patrickvonplaten 

Because the configuration process for fsmt was complex, using real defaults originally masked problems, so by not having defaults it was detecting problems of not getting the right config immediately. I'm concerned that adding defaults that look about right all kinds of unexpected problems may arise.

My reasoning is this - if you don't have the actual model to support the defaults in config, then the defaults shouldn't be like in the model - I'm talking about the parts that are very specific to a model - like languages, or vocab  sizes. 

I'm not sure I'm being successful at explaining myself.

But regardless, if you feel that the defaults you suggested make more sense - please go for it. I trust you have more insight than I do here.

Reply:
I completely agree with @stas00 even though I have gotten yelled at for the same thing ;)
Makes no sense to have defaults that look like they might actually contain useful information when they don't. Just a distraction. I have the same view on documenting config defaults in docstrings.

Reply:
Another note:  I realized that you picked the default numbers from a tiny model which wasn't truly tiny ;) I'm working on a new version that is 50 times smaller, and there the vocab size is 21, instead of 42K. PR coming.

Reply:
Hmm, interesting point of view! I guess for me it's quite natural that if you do:

```python 
config = BertConfig()
```

you would expect to get the most commonly used / standard BERT config being `bert-base-cased`, which is useful information IMO. The only other option that would make sense is to force the user to configure everything by himself, but I think putting random parameters there makes no sense. 

And since it's a big question of consistency with the library (we will not go back from having default configs I think), I see only the "real" default parameters option as the way to go. 

Would be keen to hear @sgugger and @LysandreJik opinion on that as well!

Reply:
I think @patrickvonplaten has the right intentions  following what is established across the repo. In every configuration file (e.g., [BERT](https://github.com/huggingface/transformers/blob/master/src/transformers/configuration_bert.py#L55), [ALBERT](https://github.com/huggingface/transformers/blob/master/src/transformers/configuration_albert.py#L37), [CTRL](https://github.com/huggingface/transformers/blob/master/src/transformers/configuration_ctrl.py#L30-L31)), we have defaults for configurations so that initializing one without arguments yields a sensible architecture based on existing pre-trained weights. If it is not the case for a model, then it slipped the review and should be updated.

This means that doing: `BertModel(BertConfig())` yields a model similar to what the original BERT model is. This makes it easier to work with as we don't have a myriad of (usually different) arguments to supply to each configuration when doing tests.

Also, this is the convention we have adopted until now and I see no strong argument against it that would lead to changing this convention. I also see no argument on why FSMT would need to differ in that regard.

@stas00:
> Because the configuration process for fsmt was complex, using some defaults originally masked problems, so by not having defaults it was detecting problems of not getting the right config immediately. I'm concerned that adding defaults that look about right all kinds of unexpected problems may arise.

I would say that if some problems were masked by using some defaults, then some additional tests should be added to ensure that these tests are not an issue, either for this model or for others.

Reply:
> this is the convention we have adopted until now and I see no strong argument against it that would lead to changing this convention

If you're saying that convention practically works, then it is.

> some additional tests should be added 

Yes, I have just done that here https://github.com/huggingface/transformers/pull/7860

Reply:
@LysandreJik The convention doesn't make as much sense for situations where you have many checkpoints trained from scratch, like FSMT/Marian. They all have different vocab sizes and none is meaningfully the "base" model in a `bert-base-cased` way.

Reply:
Similarly, there are two completely different blenderbot checkpoints and I arbitrarily chose the smaller to be the config defaults.

Reply:
As I mentioned twice earlier this is the way of the project, and so it is.

So I will just address @patrickvonplaten comment since i think I know why we are missing each other here:

>  I guess for me it's quite natural that if you do:
> 
> config = BertConfig()
> 
> you would expect to get the most commonly used / standard BERT config being bert-base-cased, which is useful information IMO.

I totally agree! It works for Bert and other similar models. It doesn't work for translation models, IMHO. 

What would as a user do with the default German to English translation model with 40K vocab of non-existing vocab -  the model the user will end up with will not be functional - yes, they have to configure it if they don't use a pretrained model. There is no magical way around it.

The key here is that certain models do not have sensible defaults and when this is the case giving a default that looks very much like correct default just for the consistency sake is questionable engineering-wise.

It'd work for automatic tests as in ThomWolf's recent PR as long as you don't try to do any qualitative tests, but it won't do anything useful for end users.